provision/acc_provision/acc_provision.py

```diff
@@ -950,11 +931,88 @@ def provision(args, apic_file, no_random):
     return ret
 
 
+DEFAULT_FLAVORS_PATH = os.path.dirname(os.path.realpath(__file__)) + "/flavors.yaml"
+VERSION_FIELDS = [
+    "cnideploy_version",
+    "aci_containers_host_version",
+    "opflex_agent_version",
+    "aci_containers_controller_version",
+    "openvswitch_version",
+]
+
+
+def check_yaml(text_to_verify):
+    try:
+        flavors_yaml = yaml.load(text_to_verify)
+        if 'default_flavor' in flavors_yaml:
+            return flavors_yaml
+    except yaml.YAMLError as exc:
+        print(exc)
+        raise Exception
+
+
+def default_flavors_path():
+    with open(DEFAULT_FLAVORS_PATH, 'r') as res:
+        info("Loading flavors from default flavors.yaml: " + DEFAULT_FLAVORS_PATH)
+        valid_yaml = check_yaml(res)
+        if valid_yaml:
+            set_flavors(valid_yaml)
+    return
+
+
+def get_flavors(flavors_url):
+    if flavors_url == 'blah':
+        return default_flavors_path()
+    flag = False
+    try:
+        # try as a URL
+        res = requests.get(flavors_url)
+        if res and res.status_code == 200:
+            valid_yaml = check_yaml(res.text)
+            if valid_yaml:
+                info("Loading flavors from URL: " + flavors_url)
+                set_flavors(valid_yaml)
+                flag = True
+        else:
+            raise Exception
+    except Exception:
+        # try as a local file
+        if exists(flavors_url):
+            with open(flavors_url, 'r') as res:
+                valid_yaml = check_yaml(res)
+                if valid_yaml:
+                    info("Loading flavors from local file: " + flavors_url)
+                    set_flavors(valid_yaml)
+                    flag = True
+    finally:
+        # try the static file in repo
+        if not flag:
+            return default_flavors_path()
+
+
+def set_flavors(doc):
+    global DEFAULT_FLAVOR, VERSIONS, DEFAULT_FLAVOR_OPTIONS, CfFlavorOptions, FLAVORS
+    DEFAULT_FLAVOR = doc['default_flavor']
```

Comment:
I know its not directly to this commit, but could you also update the following:
https://github.com/noironetworks/aci-containers/blob/master/provision/acc_provision/flavors.yaml#L1
to: kubernetes-1.12

Please also update the commit message accordingly.

Reply:
Modified and added in commit message